source/dmocks/mocks.d

```diff
@@ -406,6 +440,134 @@ public class ExpectationSetup
     }
 }
 
+public class TemplatedExpectationSetup(T)
+{
+    import dmocks.arguments;
+    import dmocks.expectation;
+    import dmocks.dynamic;
+    import dmocks.qualifiers;
+    import dmocks.call;
+
+    private alias Ret = ReturnType!T;
+
+    private alias Params = Parameters!T;
+
+    private CallExpectation _expectation;
+
+    private Call _setUpCall;
+
+    this(CallExpectation expectation, Call setUpCall)
+    {
+        assert(expectation !is null, "can't create an ExpectationSetup if expectation is null");
+        assert(setUpCall !is null, "can't create an ExpectationSetup if setUpCall is null");
+        _expectation = expectation;
+        _setUpCall = setUpCall;
+    }
+
+    /**
+    * Ignore method argument values in matching calls to this expectation.
```

Comment:
One space before the star, `*`, is missing here and in the comments below.
```d
/**
 * Ignore method argument values in matching calls to this expectation.
 */
```

Reply:
oops, editor fail

Reply:
done